ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/IOSimQSM/Test/StateMachine/Sequential.hs

```diff
@@ -0,0 +1,193 @@
+{-# LANGUAGE BangPatterns          #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE NamedFieldPuns        #-}
+{-# LANGUAGE RecordWildCards       #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+
+-----------------------------------------------------------------------------
+-- |
+--
+--
+-- /___This is a superficial variation of the quickcheck-state-machine___/
```

Comment:
@coot @jasagredo Have either of you considered this particular obstacle before? Any thoughts to share?

Reply:
The consensus from the Slack discussion was that `quickcheck-dynamic` is easier to use with `IOSim` since it's pure to begin with. But no one yet has much experience doing so.

The changes in this file compared to the upstream QSM are trivial, so I want to merge this as-is for now. A tech debt issue can reformulate the GSM test away from QSM into `quickcheck-dynamic`, and then delete this module.

(Or we can pushing this generalization upstream, though in the maintainer's shoes I would be hesitant to take on the necessary `io-classes` dependency.)